src/components/homepage/filter.js

```diff
@@ -1,45 +1,42 @@
-import React, { useContext, useState } from 'react';
-import { RootContext } from '../../App';
-import axios from 'axios';
-import SearchInput from './search';
-import '@fortawesome/fontawesome-free/js/fontawesome';
-import '@fortawesome/fontawesome-free/js/solid';
-import './filter.css';
-import'./feild_container.css';
+import React, { useState } from "react";
+import SearchInput from "./search";
+import "@fortawesome/fontawesome-free/js/fontawesome";
+import "@fortawesome/fontawesome-free/js/solid";
+import "./filter.css";
+import "./feild_container.css";
 
 function Filter({ countriesData, onFilterChange }) {
-    const [isOpen, setIsOpen] = useState(false);
-    const [title, setTitle] = useState("Filter by Region");
-  
+  const [isOpen, setIsOpen] = useState(false);
+  const [title, setTitle] = useState("Filter by Region");
+  const apiBaseUrl = process.env.REACT_APP_API_BASE_URL;
 
-   
   const handleSearch = (searchData) => {
     onFilterChange(searchData);
   };
 
-    function openList() {
-      setIsOpen(!isOpen);
+  function openList() {
+    setIsOpen(!isOpen);
+  }
+
+  function handleChange(e) {
+    const region = e.target.getAttribute("data-value");
+
+    if (region === "All") {
+      onFilterChange(countriesData);
+    } else {
+      fetch(`${apiBaseUrl}region/${region}`)
```

Comment:
Make an api folder into src/ directory and add an /api/ folder.
All the api calls would be written into that api.js file.

Reply:
updated